streams/src/main/java/org/apache/kafka/streams/state/internals/CompositeReadOnlySessionStore.java

```diff
@@ -41,47 +42,41 @@ public CompositeReadOnlySessionStore(final StateStoreProvider storeProvider,
         this.storeName = storeName;
     }
 
-    private interface Fetcher<K, V> {
-        KeyValueIterator<Windowed<K>, V> fetch(ReadOnlySessionStore<K, V> store);
-    }
 
-    private KeyValueIterator<Windowed<K>, V> fetch(Fetcher<K, V> fetcher) {
+    @Override
+    public KeyValueIterator<Windowed<K>, V> fetch(final K key) {
```

Comment:
For single key fetch, why we are still returning only a first satisfied match in the iterator, as compared to range query, where we return all matched items across all windows?